From e168eeef67a6680730e86ee389a383d9c1607c25 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Fri, 12 Aug 2016 11:24:47 +0200
Subject: [PATCH] systemd/dracut-cmdline.sh: unset "UNSET" $root

https://bugzilla.redhat.com/show_bug.cgi?id=1365667
caused by commit 9aa224cc

(cherry picked from commit fc77aca7ddbd33328066a943a4de09de34d25c4c)
---
 modules.d/98dracut-systemd/dracut-cmdline.sh | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules.d/98dracut-systemd/dracut-cmdline.sh b/modules.d/98dracut-systemd/dracut-cmdline.sh
index ccf24fb..6c6ee02 100755
--- a/modules.d/98dracut-systemd/dracut-cmdline.sh
+++ b/modules.d/98dracut-systemd/dracut-cmdline.sh
@@ -25,7 +25,7 @@ source_conf /etc/conf.d
 # between the case where it was set to the empty string and the case where it
 # wasn't specified at all.
 if ! root="$(getarg root=)"; then
-    root='UNSET'
+    root_unset='UNSET'
 fi
 
 rflags="$(getarg rootflags=)"
@@ -49,7 +49,7 @@ source_hook cmdline
 
 [ -f /lib/dracut/parse-resume.sh ] && . /lib/dracut/parse-resume.sh
 
-case "$root" in
+case "${root}${root_unset}" in
     block:LABEL=*|LABEL=*)
         root="${root#block:}"
         root="$(echo $root | sed 's,/,\\x2f,g')"
@@ -75,7 +75,7 @@ case "$root" in
         rootok=1 ;;
 esac
 
-[ -z "$root" ] && die "Empty root= argument"
+[ -z "${root}${root_unset}" ] && die "Empty root= argument"
 [ -z "$rootok" ] && die "Don't know how to handle 'root=$root'"
 
 export root rflags fstype netroot NEWROOT
